### PR TITLE
Migrate service_extensions.dart to null safety.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

File: packages/devtools_app/lib/src/shared/service_extensions.dart [hunk 1]
```diff
@@ -91,19 +91,20 @@ class ServiceExtensionDescription<T> {
 
   final List<String> displayValues;
 
-  final String gaScreenName; // Analytics screen (screen name where item lives).
+  final String?
+      gaScreenName; // Analytics screen (screen name where item lives).
 
-  final String gaItem; // Analytics item name (toggleable item's name).
+  final String? gaItem; // Analytics item name (toggleable item's name).
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/service_extensions.dart:97]: formatting. For both of these, move the documentation to be above the class members and change to proper dart doc with three slashes ///

File: packages/devtools_app/lib/src/shared/service_extensions.dart [hunk 2]
```diff
@@ -502,7 +503,7 @@ final List<ServiceExtensionDescription> _extensionDescriptions = [
   disablePhysicalShapeLayers,
 ];
 
-final Map<String, ServiceExtensionDescription> serviceExtensionsAllowlist =
+final Map<String?, ServiceExtensionDescription?> serviceExtensionsAllowlist =
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/service_extensions.dart:506]: these both can be non-nullable

File: packages/devtools_app/lib/src/shared/service_extensions.dart [hunk 3]
```diff
@@ -47,9 +47,9 @@ class ToggleableServiceExtensionDescription<T>
 
   static const disabledValueIndex = 1;
 
-  T get enabledValue => values[enabledValueIndex];
+  T? get enabledValue => values[enabledValueIndex];
 
-  T get disabledValue => values[disabledValueIndex];
+  T? get disabledValue => values[disabledValueIndex];
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/shared/service_extensions.dart]: done

File: packages/devtools_app/lib/src/shared/service_extensions.dart (hunk 1 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/shared/service_extensions.dart:97]: done

File: packages/devtools_app/lib/src/shared/service_extensions.dart (hunk 2 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/shared/service_extensions.dart]: done

**Review by kenzieschmoll — APPROVED** [review]

**Comment by polina-c** [comment]
tested

**Pull request merged**